MIP/mip-n/README.md

```diff
@@ -0,0 +1,148 @@
+- `<release-type>`: the type of release, e.g., `stable`, `beta`, or `rc`.
+
+Note that `<release-type>` is ill-defined. Individual software units may define their own release types.
+
+Note further that we do not specify environments such as `devnet`, `testnet`, or `mainnet` in the release name. Though these may be specified in the `<release-type>` field, we recommend that the governance for each software provide indirection from the environment to a given release. 
+
+We thus propose the following standard for releases marking the canonical release which should be used in a given environment:
+
+```
+<software-unit>-<environment>
+```
+
+An example of how this can be released in Git and GitHub is provided in the following section.
+
+### Standardized Source Control
+1. Source control MUST be responsible for initiating all releases with the following tags:
+    1. A tag corresponding to the Standardized Release Naming Convention above.
+    2. A tag corresponding to the commit hash of the release.
+    3. A tag corresponding to the environment indirection for the release should it be valid.
+
+2. Releases corresponding to environment indirection MUST only be tagged by making a release from the default branch of the repository.
+
+3. By default, the `<release-type>` SHOULD correspond to the branch from which the release is made.
+
+4. Tag indirection CAN be performed as follows:
+
+```bash
+git tag v1.0.0
+git push origin v1.0.0
+git tag -f latest v1.0.0 
+git push origin -f latest 
+```
+
+5. GitHub SHOULD be the default source control platform. Within the platform each release should contain at least the archive of the source code which would produce the release artifact.
+
+6. Well-known configuration files MUST also be included in the release archive where they apply. 
+
+7. Releases outside of GitHub, e.g., DockerHub, MUST be tagged with the same tags as the GitHub release and linked in the release notes on the GitHub release.
+
+8. Releases to established environments MUST follow governance guidelines for the software unit in question.
+
+9. Releases to established environments MUST trigger an update of all usage of that software unit to the appropriate tag by Movement Labs. This SHOULD be automated to the extent possible to ensure that the source control reflects the canonical release.
+
+### Programmatic Access to Release Artifacts
+1. Source control SHOULD be the primary source of release artifacts. 
+
+2. Artifacts which are in consumable forms such as container images, binaries, or libraries can be consumed as the developer sees fit.
+
+3. We recommend the following Rust pseudocode API for fetching artifacts such as well-known configuration files, contract addresses, or other artifacts which may need an application context to be properly consumed:
+
+```rust
+trait Release<T> {
+    fn fetch_artifact(&self) -> Result<T, Error>;
+}
```

Comment:
I don't see a use case for this being a trait, generalizing over different types of things to fetch.
File downloads and metadata fetches are rather different, what piece of code would need to generalize `fetch_artifact`?

Reply:
This is intended as pseudocode describing different artifacts that could be fetched for different release types. But, what would you suggest?

Reply:
I think I understand the intent. Maybe there should be a text paragraph suggesting that the actual API does not have to be this generic, feature an unifying trait, etc.